packages/create-release/src/helpers.ts

```diff
@@ -0,0 +1,270 @@
+  // Check if the composer.json file exists.
+  if (fs.existsSync(`${path}/composer.json`)) {
+    try {
+      const contents = JSON.parse(
+        fs.readFileSync(`${path}/composer.json`, 'utf8'),
+      );
+
+      if (contents.version) {
+        if (valid(contents.version)) {
+          return contents.version;
+        }
+
+        exitError(
+          `The version number in the composer.json file is not a valid semver version number: ${contents.version}`,
+        );
+      }
+    } catch (error) {
+      // Do nothing.
+    }
+  }
+
+  // Retrieve the plugin file to read the headers from.
+  const files = [
+    `${path.split('/').pop()}.php`,
+    'plugin.php',
+  ];
+
+  // Loop through the files and check if they exist.
+  for (let i = 0; i < files.length; i += 1) {
+    const file = `${path}/${files[i]}`;
+
+    if (!fs.existsSync(file)) {
+      continue; // eslint-disable-line no-continue
+    }
+
+    const headers = readPluginHeaders(file);
+
+    if (headers?.version) {
+      if (valid(headers.version)) {
+        return headers.version;
+      }
+
+      exitError(
+        `The version number in the plugin file is not a valid semver version number: ${headers.version}`, // eslint-disable-line max-len
+      );
+    }
+  }
+
+  return undefined;
+}
+
+/**
+ * Retrieve the release type from the current version and the new version.
+ */
+function getReleaseType(currentVersion: string, newVersion: string): ReleaseType | undefined {
+  const currentParts = currentVersion.split('.');
+  const newParts = newVersion.split('.');
+
+  if (newParts.length !== 3) {
+    return undefined;
+  }
+
+  if (currentParts.length !== 3) {
+    return 'major';
+  }
+
+  if (newParts[0] > currentParts[0]) {
+    return 'major';
+  }
+
+  if (newParts[1] > currentParts[1]) {
+    return 'minor';
+  }
+
+  if (newParts[2] > currentParts[2]) {
+    return 'patch';
+  }
+
+  return undefined;
+}
+
+/**
+ * Bump the version number given the type of release.
+ */
+function bumpVersion(version: string, releaseType: ReleaseType): string {
+  const versionParts = version.split('.');
+  const major = parseInt(versionParts[0], 10);
+  const minor = parseInt(versionParts[1], 10);
+  const patch = parseInt(versionParts[2], 10);
+
+  switch (releaseType) {
+    case 'major':
+      return `${major + 1}.0.0`;
+    case 'minor':
+      return `${major}.${minor + 1}.0`;
+    case 'patch':
+      return `${major}.${minor}.${patch + 1}`;
+    default:
+      return version;
+  }
+}
+
+/**
+ * Upgrade the version number in the composer.json file.
+ */
+function upgradeComposerVersion(basePath: string, version: string): void {
+  try {
+    const contents = JSON.parse(
+      fs.readFileSync(`${basePath}/composer.json`, 'utf8'),
+    );
+
+    // Insert the version after the description/name index.
+    const insertIndex = Object.keys(contents).indexOf('description') || Object.keys(contents).indexOf('name');
```

Comment:
what if `version` already exists? Am I missing something here?

Reply:
Let me test for that, good call